File: codes/test/test/quicksort.swift
var randomNumbers = [42, 12, 88, 62, 63, 56, 1, 77, 88, 97, 97, 20, 45, 91, 62, 2, 15, 31, 59, 5]

func quickSort(array: [Int]) -> [Int] {
    var less = [Int]()
    var equal = [Int]()
    var greater = [Int]()
    
    if array.count > 1 {
        let pivot = array[0]
        
        for x in array {
            if x < pivot {
                less.append(x)
            } else if x == pivot {
                equal.append(x)
            } else {
                greater.append(x)
            }
        }
        
        return (quickSort(array: less) + equal + quickSort(array: greater))
    } else {
        return array
    }
}

print(quickSort(array: randomNumbers))

File: codes/test/result-cpp/quicksort.cpp
#include "helper.cpp"
vector<int> randomNumbers = {42, 12, 88, 62, 63, 56, 1, 77, 88, 97, 97, 20, 45, 91, 62, 2, 15, 31, 59, 5};
vector<int> quickSort(vector<int> array)
{
    vector<int> less = vector<int>();
    vector<int> equal = vector<int>();
    vector<int> greater = vector<int>();
    if (array.size() > 1)
    {
        const int pivot = array[0];
        for (vector<int>::iterator it = array.begin(); it != array.end(); ++it)
        {
            int x = *it;
            if (x < pivot)
            {
                less.push_back(x);
            }
            else
            if (x == pivot)
            {
                equal.push_back(x);
            }
            else
            {
                greater.push_back(x);
            }
        }
        return (quickSort(less) + equal + quickSort(greater));
    }
    else
    {
        return array;
    }
}
int main() {
    print(quickSort(randomNumbers));
    return 0;
}
